### PR TITLE
Use new severity levels (high, medium, low)

### PR DESCRIPTION
**Pull request by NIGCH**
A tentative fix for https://github.com/dependency-check/dependency-check-sonar-plugin/issues/870. 

A change to DependencyCheckUtils.severityToScore() also appears to be needed to always use the [CVSS scores in the specification](https://www.first.org/cvss/v3.0/specification-document#Qualitative-Severity-Rating-Scale) to map a severity string to a CVSS score, rather than the configurable thresholds which determine how CVSS scores map to SonarQube severities. However, this will be addressed in a later pull request.

### CODE REVIEW

File: sonar-dependency-check-plugin/src/main/java/org/sonar/dependencycheck/rule/KnownCveRuleDefinition.java [hunk 1]
```diff
@@ -42,14 +42,13 @@ public void define(Context context) {
         fillOWASPRule(rule);
         NewRule ruleWithSecurityHotspot = repo.createRule(DependencyCheckConstants.RULE_KEY_WITH_SECURITY_HOTSPOT);
         fillOWASPRule(ruleWithSecurityHotspot);
-        ruleWithSecurityHotspot.setType(RuleType.SECURITY_HOTSPOT);
```
Comment (Reamer) [inline sonar-dependency-check-plugin/src/main/java/org/sonar/dependencycheck/rule/KnownCveRuleDefinition.java:45]: If Security Hotspot Rules are deprecated, we can drop the additional rule.

File: README.md [hunk 2]
```diff
@@ -98,10 +98,10 @@ This plugin tries to add SonarQube issues to your project configuration files (e
 To configure the severity of the created issues you can optionally specify the minimum score for each severity with the following parameter. Specify a score of `-1` to completely disable a severity.
 
 ```ini
-sonar.dependencyCheck.severity.blocker=9.0
-sonar.dependencyCheck.severity.critical=7.0
-sonar.dependencyCheck.severity.major=4.0
-sonar.dependencyCheck.severity.minor=0.0
+sonar.dependencyCheck.severity.critical=9.0
```
Comment (Reamer) [inline README.md:101]: This refers to the SonarQube severity level. Critical should not be here.

File: sonar-dependency-check-plugin/src/main/java/org/sonar/dependencycheck/base/DependencyCheckConstants.java [hunk 3]
```diff
@@ -23,19 +23,19 @@ public final class DependencyCheckConstants {
 
     public static final String JSON_REPORT_PATH_PROPERTY = "sonar.dependencyCheck.jsonReportPath";
     public static final String HTML_REPORT_PATH_PROPERTY = "sonar.dependencyCheck.htmlReportPath";
-    public static final String SEVERITY_BLOCKER = "sonar.dependencyCheck.severity.blocker";
     public static final String SEVERITY_CRITICAL = "sonar.dependencyCheck.severity.critical";
```
Comment (Reamer) [inline sonar-dependency-check-plugin/src/main/java/org/sonar/dependencycheck/base/DependencyCheckConstants.java:26]: Critical is no longer used by SonarQube.

**Review by Reamer — CHANGES_REQUESTED** [review]
Thank you very much for the pull request. This is quite a complicated pull request. Thank you for splitting it up. I haven't looked at these lines of code for a long time.

I have added a few comments where we need to take a look.

btw. Table entry in "Plugin version compatibility" (Readme.md) is missing.

**Comment by Reamer** [comment]
Perhaps I was too hasty with the request to split the pull request.

I think it is necessary that the constants are also required in this pull request.
https://github.com/dependency-check/dependency-check-sonar-plugin/pull/894/files#diff-ce8ce310fb93461c861183d57d10967a8bf3d7a2de59aa4053436ef5ef661817R38-R41

It seems that in the current code the score of CVSS and SonarQube has been mixed.

**Comment by Reamer** [comment]
I think that makes sense, doesn't it?

File: README.md (hunk 2 above)
Comment (NIGCH) [reply README.md:101]: Agreed. It was left in until the CVSS scores were disentangled from the SonarQube scores.

File: sonar-dependency-check-plugin/src/main/java/org/sonar/dependencycheck/base/DependencyCheckConstants.java (hunk 3 above)
Comment (NIGCH) [reply sonar-dependency-check-plugin/src/main/java/org/sonar/dependencycheck/base/DependencyCheckConstants.java]: Agreed, as above.

**Comment by NIGCH** [comment]
> I think that makes sense, doesn't it?

Yes, thanks, it looks good to me. Is the compatibility entry I've added correct?

File: sonar-dependency-check-plugin/src/main/java/org/sonar/dependencycheck/DependencyCheckConfiguration.java [hunk 4]
```diff
@@ -48,37 +48,32 @@ public static List<PropertyDefinition> getPropertyDefinitions() {
                         .description("path to the 'dependency-check-report.html' file")
                         .defaultValue(DependencyCheckConstants.HTML_REPORT_PATH_DEFAULT)
                         .build(),
-                PropertyDefinition.builder(DependencyCheckConstants.SEVERITY_BLOCKER)
+                PropertyDefinition.builder(DependencyCheckConstants.SEVERITY_HIGH)
```
Comment (Reamer) [inline sonar-dependency-check-plugin/src/main/java/org/sonar/dependencycheck/DependencyCheckConfiguration.java:51]: Please set the configuration option `deprecatedKey` to the old Critical configuration key.

File: sonar-dependency-check-plugin/src/main/java/org/sonar/dependencycheck/DependencyCheckConfiguration.java [hunk 5]
```diff
@@ -48,37 +48,32 @@ public static List<PropertyDefinition> getPropertyDefinitions() {
...
-                        .description("Minimum score for blocker issues or -1 to deactivate blocker issues.")
-                        .defaultValue("9.0")
+                        .name("High")
+                        .description("Minimum score for high issues or -1 to deactivate high issues.")
+                        .defaultValue(Float.toString(DependencyCheckConstants.SEVERITY_HIGH_DEFAULT))
                         .type(PropertyType.FLOAT)
+                        .index(1)
                         .build(),
-                PropertyDefinition.builder(DependencyCheckConstants.SEVERITY_CRITICAL)
+                PropertyDefinition.builder(DependencyCheckConstants.SEVERITY_MEDIUM)
```
Comment (Reamer) [inline sonar-dependency-check-plugin/src/main/java/org/sonar/dependencycheck/DependencyCheckConfiguration.java:60]: Please set the configuration option `deprecatedKey` to the old Major configuration key.

File: sonar-dependency-check-plugin/src/main/java/org/sonar/dependencycheck/DependencyCheckConfiguration.java [hunk 6]
```diff
@@ -48,37 +48,32 @@ public static List<PropertyDefinition> getPropertyDefinitions() {
...
-                        .description("Minimum score for critical issues or -1 to deactivate critical issues.")
-                        .defaultValue(Float.toString(DependencyCheckConstants.SEVERITY_CRITICAL_DEFAULT))
+                        .name("Medium")
+                        .description("Minimum score for medium issues or -1 to deactivate medium issues.")
+                        .defaultValue(Float.toString(DependencyCheckConstants.SEVERITY_MEDIUM_DEFAULT))
                         .type(PropertyType.FLOAT)
+                        .index(2)
                         .build(),
-                PropertyDefinition.builder(DependencyCheckConstants.SEVERITY_MAJOR)
+                PropertyDefinition.builder(DependencyCheckConstants.SEVERITY_LOW)
```
Comment (Reamer) [inline sonar-dependency-check-plugin/src/main/java/org/sonar/dependencycheck/DependencyCheckConfiguration.java:69]: Please set the configuration option `deprecatedKey` to the old Minor configuration key.

**Comment by Reamer** [comment]
> > I think that makes sense, doesn't it?
> 
> Yes, thanks, it looks good to me. Is the compatibility entry I've added correct?

Looks good.

**Review by Reamer — APPROVED** [review]
LGTM

**Comment by NIGCH** [comment]
Thank you @Reamer - much appreciated.

**Comment by NIGCH** [comment]
No, that's everything.

**Pull request merged**